distributed/metrics.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import print_function, division, absolute_import
+
+import collections
+
+
+_empty_namedtuple = collections.namedtuple("_empty_namedtuple", ())
+
+
+def _run_psutil(method_name, default):
+    try:
+        import psutil
+    except ImportError:
+        return default
+    try:
+        return getattr(psutil, method_name)()
+    except RuntimeError:
+        # This can happen on some systems (e.g. no physical disk in worker)
+        return default
```

Comment:
Perhaps we should move the psutil check outside?

```python
try:
    from psutil import disk_io_counters, net_io_counters
except importError:
    disk_io_counters = net_io_counters = _empty_named_tuple
```

Given the comment on no physical disk I suspect that the solution above may be overly simplistic.  Perhaps we could try running each method once here?